integrations/huggingface/src/actions/generateContent.ts

```diff
@@ -0,0 +1,120 @@
+import * as sdk from "@botpress/sdk";
+import { interfaces } from "@botpress/sdk";
+import type { HfInference } from "@huggingface/inference";
+import { InvalidPayloadError } from "@botpress/client";
+
+type HfToolCalls = Awaited<
+  ReturnType<HfInference["chatCompletion"]>
+>["choices"][number]["message"]["tool_calls"];
+
+function mapToolCalls(
+  hfToolCalls: HfToolCalls
+): sdk.interfaces.llm.ToolCall[] | undefined {
+  return hfToolCalls
+    ?.filter((toolCall) => toolCall.type === "function")
+    .map((toolCall) => ({
+      id: `${toolCall.id}`,
+      type: toolCall.type as "function",
+      function: {
+        name: toolCall.function.name,
+        arguments: toolCall.function.arguments as Record<string, any> | null,
+      },
+    }));
+}
+
+function mapToStopReason(hfFinishReason: string) {
+  switch (hfFinishReason) {
```

Comment:
this might be more readable with a hashmap (albeit a bit slower)

Reply:
I went with switch case because it successfully infers return types as string literals, opposed to map which infers it as a generic string type.<img width="574" alt="Screenshot 2024-08-29 at 11 50 57 AM" src="https://github.com/user-attachments/assets/4bf6c37c-9e55-4c51-9d13-a508f13e7665">
<img width="551" alt="Screenshot 2024-08-29 at 11 51 45 AM" src="https://github.com/user-attachments/assets/85d6928a-b9ad-490d-bba5-19d6acd92bb6">